--- tests/test_multiprocessing.py.orig	2016-03-14 07:03:46.071578677 +0000
+++ tests/test_multiprocessing.py	2016-03-14 07:04:45.061584576 +0000
@@ -1,4 +1,5 @@
 import unittest
+import time
 try:
     from multiprocessing import Process, Queue, Pipe
     disabled = False
@@ -19,6 +20,7 @@ class test_ini(unittest.TestCase):
         q = Queue()
         w = Queue()
         q.put(cfg)
+        time.sleep(2)
         p = Process(target=getxy, args=(q, w))
         p.start()
         self.assertEqual(w.get(timeout=1), '42')
